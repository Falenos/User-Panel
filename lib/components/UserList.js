import React from 'react';
import User from './User';

const UserList = (props) => {
  return (
    <div>
      {Object.values(props.users).map(user =>
        <User
          key = {user.id}
          user = {user}
        />
      )}
    </div>
  );
};

export default UserList;
