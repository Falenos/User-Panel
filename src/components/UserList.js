import React from 'react';
import PropTypes from 'prop-types';
import User from './User';
import { connect } from 'react-redux';

class UserList extends React.Component {

  render() {
    return (
      <div className='grid'>
        {Object.values(this.props.users).map(user =>
          <User
            key={user.id}
            user={user}
          />
        )}
      </div>
    );
  }
}

UserList.propTypes = {
  user: PropTypes.object
};

export default connect(
  (state) => {
    return {
      users: state.users
    };
  }
)(UserList);
