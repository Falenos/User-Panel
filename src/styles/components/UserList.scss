.grid {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  // to avoid streching of row items
  align-items: flex-start;
}
.grid ~ .overlay {
  height: 100vh;
}
