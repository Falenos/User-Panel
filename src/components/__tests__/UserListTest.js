import React from 'react';
import UserList from '../UserList';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configure from '../../store/config';


describe('UserList', () => {

  const testProps = {
    users: {
      a: {id: 'a'},
      b: {id: 'b'}
    }
  };
  const store = configure(testProps);

  it('renders correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <UserList />
      </Provider>
    ).toJSON();

    expect(tree.children.length).toBe(2);

    expect(tree).toMatchSnapshot();
  });

});
