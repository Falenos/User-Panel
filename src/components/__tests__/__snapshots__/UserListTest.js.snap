// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList renders correctly 1`] = `
<div
  className="grid"
>
  <div
    className="grid-item"
    onClick={[Function]}
  >
    <div
      className="grid-item-header"
    >
      <div>
        a
      </div>
    </div>
    <div
      className="grid-item-content"
    >
      <div>
        a
      </div>
      <div>
        a
      </div>
      <div>
        a
      </div>
      <div>
        a
      </div>
      <div>
        a
      </div>
    </div>
  </div>
  <div
    className="grid-item"
    onClick={[Function]}
  >
    <div
      className="grid-item-header"
    >
      <div>
        b
      </div>
    </div>
    <div
      className="grid-item-content"
    >
      <div>
        b
      </div>
      <div>
        b
      </div>
      <div>
        b
      </div>
      <div>
        b
      </div>
      <div>
        b
      </div>
    </div>
  </div>
</div>
`;
